sdk/client.go

```diff
@@ -83,3 +85,19 @@ func (c *Client) getServiceStatus(resPath string) (*api.MetaMessage, error) {
 	}
 	return msg, nil
 }
+
+func (c *Client) postServiceAction(resPath string, req interface{}) error {
+	resp, resErr := c.postResource(resPath, req)
+	// Since the response body can have more information about the error, we try to parse it
+	if resErr != nil && resp == nil {
+		return fmt.Errorf("postResource: %w", resErr)
+	}
+	msg := api.MetaMessage{}
+
+	if err := json.Unmarshal(resp, &msg); err != nil {
+		// if the response is not a MetaMessage, it's probably a success message
+		// Therefore we just return the original error from the request
+		return resErr
```

Comment:
Why is the success message in the response error?

Reply:
The main reason is sometimes the response message is not in the `MetaMessage` format therefore the unmarshal fails and there might be simply an http error like could not find the server, etc.. so we return the response error in that case.
I think it need to have more clear comment.
or maybe we change it to a better way. do you have any suggestions in mind?

Reply:
pushed a new change, hope the new one is more clear. let me know plz

Reply:
I got it now. Thank you for clarifying :+1: